x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/hooks/form.tsx

```diff
@@ -391,7 +391,7 @@ export function useOnSubmit({
 
       // Check if agentless is configured in ESS and Serverless until Agentless API migrates to Serverless
       const isAgentlessConfigured =
-        isAgentlessAgentPolicy(createdPolicy) || isAgentlessPackagePolicy(data!.item);
+        isAgentlessAgentPolicy(createdPolicy) || (data && isAgentlessPackagePolicy(data.item));
```

Comment:
when `data` is undefined, this line previously prevented `error` from being handled and bubbled up to the UI

cc @Omolola-Akinleye